api/src/middleware/rate-limiter.ts

```diff
@@ -0,0 +1,64 @@
+/**
+ *  RateLimiter using Redis
+ *  and rate-limiter-flexible
+ *  can extend with further options
+ *  in future
+ */
+import { RequestHandler } from 'express';
+import redis from 'redis';
+import { RateLimiterRedis } from 'rate-limiter-flexible';
+import { HitRateLimitException } from '../exceptions';
+import { RedisNotFoundException } from '../exceptions';
+import env from '../env';
+
+const redisClient = redis.createClient({
+	enable_offline_queue: false,
+	host: env.REDIS_HOST,
+	port: env.REDIS_PORT,
+	password: env.REDIS_PASSWORD,
+});
+
+const rateLimiter: RequestHandler = (req, res, next) => {
```

Comment:
Change this to an async/await function:

```suggestion
const rateLimiter: RequestHandler = asyncHandler(async (req, res, next) => {
```

Reply:
(async handler comes from `import asyncHandler from 'express-async-handler'`)

Reply:
Sounds good to me 👍